### ISSUE TITLE
The type ... exists in both 'System.Drawing.Common, Version=4.0.0.1, ... and 'System.Drawing, Version=4.0.0.0, ...

### ISSUE THREAD

**Issue by maettu-this** [swharden/ScottPlot#655]
I have a WinForms application on .NET 4.8 that implements user controls, i.e. references various types of `System.Drawing`.

ScottPlot has a dependency to `System.Drawing.Common`:

![Dependency to System Drawing Common](https://user-images.githubusercontent.com/24897735/101759279-ad9d1800-3ad9-11eb-96b3-f112c933c925.png)

Adding ScottPlot to the existing application results in:
```
Error      CS0433  The type 'Bitmap' exists in both 'System.Drawing.Common, Version=4.0.0.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' and 'System.Drawing, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
Error      CS0433  The type 'SolidBrush' exists in both 'System.Drawing.Common, Version=4.0.0.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' and 'System.Drawing, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
Error      CS0433  The type 'Pen' exists in both 'System.Drawing.Common, Version=4.0.0.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' and 'System.Drawing, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
Error      CS0433  The type 'Pen' exists in both 'System.Drawing.Common, Version=4.0.0.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' and 'System.Drawing, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
Error      CS0433  The type 'Icon' exists in both 'System.Drawing.Common, Version=4.0.0.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' and 'System.Drawing, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
```

How can I overcome this mismatch? Is there an older version without the dependency to `System.Drawing.Common` I could use? Or could `ScottPlot.dll` be specifically built for `net472` without a dependency to `System.Drawing.Common`?

**Comment by swharden** [comment]
Hi @maettu-this, sorry it took so long to get back to you on this.

ScottPlot references `System.Drawing.Common` which is available on NuGet and intended to replace `System.Drawing` references. My guess is your project has a reference to `System.Drawing`. I think you can fix it by removing that reference and installing `System.Drawing.Common` instead. Once `System.Drawing.Common` can be installed, installing `ScottPlot` should be easy.

If you have trouble you can look at your csproj file closely to see if there are any references to the old System.Drawing. This is much easier to do with modern SDK style csproj files. Depending on your application, you might find it worthwhile to make the change to a SDK style csproj file anyway https://www.hanselman.com/blog/upgrading-an-existing-net-project-files-to-the-lean-new-csproj-format-from-net-core

Hope it helps!
Let me know if you still have an issue 👍

**Comment by maettu-this** [comment]
Removing `System.Drawing` is not the solution:
```
Error      CS0246  The type or namespace name 'Point' could not be found (are you missing a using directive or an assembly reference?)
Error      CS0246  The type or namespace name 'Color' could not be found (are you missing a using directive or an assembly reference?)
```
This is why I earlier asked:

> Is there an older version without the dependency to System.Drawing.Common I could use?

If you can tell me the least recent version without the dependency to `System.Drawing.Common`, I could fall back to that version. This would be a quick solution.

> Or could ScottPlot.dll be specifically built for net472 without a dependency to System.Drawing.Common?

This would be the proper solution in my opinion. While there might be technical ways to use `System.Drawing.Common` in .NET Framework, e.g. https://stackoverflow.com/questions/54984173/how-do-i-install-system-drawing-common-in-a-project-that-uses-netframework-ver, the assembly got created for .NET Core, where it of course is to be used. But it should not be a requirement for a .NET Framework library.

**Comment by swharden** [comment]
> Removing System.Drawing is not the solution:

If you remove the `System.Drawing` reference you must install the `System.Drawing.Common` NuGet package to replace it

`System.Drawing.Common` supplies you with `Point`, `Color`, etc. 👍 

> least recent version without the dependency to System.Drawing.Common, I could fall back to that version

ScottPlot 3.1.6 is the most recent version that does not use `System.Drawing.Common`. It uses the .NET Framework reference to `System.Drawing`, meaning the library cannot be used on Linux, MacOS, or even on Windows in .NET Core / .NET 5 projects. I don't recommend using this old version, but it's available on NuGet https://www.nuget.org/packages/ScottPlot/3.1.6

Hope it helps!
Scott

**Comment by maettu-this** [comment]
Of course I added the `System.Drawing.Common` assembly instead, after all that was the goal. Tomorrow, I'll compose an example demonstrating the issue and attach it here. Still, the truly proper solution is to build ScottPlot in a clean way for .NET Framework, and that is without `System.Drawing.Common`. But I don't want to modify ScottPlot code and then build the assembly myself, I prefer using "official" packages. Thus, if I cannot get this issue fixed, I will go with the 3.1.6 version for a while. Not ideal, but less hassle.

**Comment by swharden** [comment]
Hey @maettu-this, thanks for looking into it further! 

If you include a zip of your project would it would be helpful in troubleshooting this issue. I bet you're not the only person who has run into this issue, so figuring out how to fix it would be helpful for future readers as well.

I opened Visual Studio, created a new .NET Framework 4.8 WinForms project, installed the ScottPlot package, then drag/dropped a FormsPlot onto the window. It seems to work fine on my end, and you can download this project here: [WindowsFormsApp18.zip](https://github.com/swharden/ScottPlot/files/5726238/WindowsFormsApp18.zip) (you'll need to "rebuild" the project after opening the solution)

![image](https://user-images.githubusercontent.com/4165489/102817124-2be0a080-439d-11eb-86b1-a2e32eca77e6.png)
